### PR TITLE
(#116) added net5.0+ TFM to match IsNetCore

### LINKED ISSUE #116: dotnet 5 not recognized as netcore
`net5.0` should be parsed as a netcore target framework instead of a netframework target framework. 

In all of the codepaths that include a conditional on `IsNetCore`, the wrong branch is taken for `net5.0` trying to treat it as net framework csproj file.

https://github.com/cake-contrib/Cake.Incubator/blob/ccccbc0ae68a3a8156769286eee1daab9a7989da/src/Cake.Incubator/Project/ProjectParserExtensions.cs#L34-L36

### PR DESCRIPTION
**Pull request by nils-a**
## Description
* Simplified the RegEx to parse TFM
* added a RegEx to match `net5.0`+ TFM as `netCore`

## Related Issue
fixes #116

## Motivation and Context
#116

## How Has This Been Tested?
I added unit tests for the existing RegEx and the added one.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/Cake.Incubator/Project/ProjectParserExtensions.cs [hunk 1]
```diff
@@ -31,10 +31,12 @@ namespace Cake.Incubator.Project
...
     {
-        private static readonly Regex NetCoreTargetFrameworkRegex = new Regex("([Nn][Ee][Tt])([Cc])\\w+", RegexOptions.Compiled);
-        private static readonly Regex NetStandardTargetFrameworkRegex = new Regex("([Nn][Ee][Tt])([Ss])\\w+", RegexOptions.Compiled);
-        private static readonly Regex NetFrameworkTargetFrameworkRegex = new Regex("([Nn][Ee][Tt][0-9*])\\w+", RegexOptions.Compiled);
-
+        private const RegexOptions parseOptions = RegexOptions.Compiled | RegexOptions.IgnoreCase;
+        private static readonly Regex NetCoreTargetFrameworkRegex = new Regex("^netcoreapp\\d+\\.\\d+$", parseOptions);
+        private static readonly Regex NetStandardTargetFrameworkRegex = new Regex("^netstandard\\d+\\.\\d+$", parseOptions);
+        private static readonly Regex NetFrameworkTargetFrameworkRegex = new Regex("^net\\d+$", parseOptions);
+        private static readonly Regex Net5PlusTargetFrameworkRegex = new Regex("^net\\d+\\.\\d+(-\\w+)?$", parseOptions);
```
Comment (mryanmurphy) [inline src/Cake.Incubator/Project/ProjectParserExtensions.cs:38]: The target framework OS identifier may also include a version, example in the docs shows `net6.0-ios14.0` that wouldn't match `^net\d+\.\d+(-\w+)?$`. 

```suggestion
        private static readonly Regex Net5PlusTargetFrameworkRegex = new Regex("^net\\d+\\.\\d+(-[\\w.]+)?$", parseOptions);
```

https://docs.microsoft.com/en-us/dotnet/standard/frameworks#os-version-in-tfms

Comment (nils-a) [reply src/Cake.Incubator/Project/ProjectParserExtensions.cs]: Thanks @mryanmurphy, good catch.

**Review by gep13 — APPROVED** [review]
LGTM!

**Comment by gep13** [comment]
@nils-a thanks for getting this fixed up!

**Pull request merged**